src/cci/cas_cci.c

```diff
@@ -712,14 +712,29 @@ cci_disconnect (int mapped_conn_id, T_CCI_ERROR * err_buf)
 
       get_last_error (con_handle, err_buf);
     }
-  else if (con_handle->broker_info[BROKER_INFO_CCI_PCONNECT] && hm_put_con_to_pool (con_handle->id) >= 0)
+  else if (con_handle->broker_info[BROKER_INFO_CCI_PCONNECT] && hm_put_con_to_pool_exceed () == 0)
     {
-      cci_end_tran_internal (con_handle, CCI_TRAN_ROLLBACK);
-      API_ELOG (con_handle, 0);
+      MUTEX_LOCK (con_handle_table_mutex);
+      if (con_handle->id < 0 || hm_put_con_to_pool (con_handle->id) < 0)
```

Comment:
qe_con_close and cci_end_tran_internal has different functions, so I split the flow not to occur unexpected error